### PR TITLE
Consider folders with a single .DS_Store file inside them to be empty.

### PR DESCRIPTION
**Pull request by nightmode**
This option is for any OS X users who are wondering why their seemingly empty folders are not being deleted. The culprit is most likely hidden .DS_Store files which the code will now check for.

Basically, with options.folders and options.dsStores set to true, remove any folders with lone .DS_Store files inside them. :)

### CODE REVIEW

**Comment by stevenvachon** [comment]
I think it'd be better to use the [junk](https://www.npmjs.com/package/junk) package.

**Comment by ForestMist** [comment]
Ah, so instead of checking for one kind of junk file (.DS_Store) check to see if there are one or more junk files. Assuming all remaining files in a folder are junk, remove that folder ya?

If yes, shall I code that up and submit an updated pull or would you prefer people use the junk library on their own and keep grunt-cleanempty as is?

**Comment by stevenvachon** [comment]
Sure, go for it. I agree with the idea and your PR, I just think that hitting all junk files would be more complete.

**Comment by stevenvachon** [comment]
Also, I'd be tempted to default `dsStores` to `true`. What do you think?

**Comment by stevenvachon** [comment]
Oh, and rename `dsStores` to `junk` or similar

**Comment by ForestMist** [comment]
New code posted.

If options.noJunk is set to true, it will use the junk library to check if a folder contains nothing but junk files. It it does, that folder will be considered empty and removed. Well assuming options.folders is also set to true of course.

As for a default true value for options.noJunk, I'm not sure. If existing users expect a certain behavior this may change that. With options.noJunk set to true things will be cleaner by default. With options.noJunk set to false things will be safer by default. Could flip a coin or roll some dice to decide, hehe.

Let me know if you see anything you want tweaked. :D

**Comment by stevenvachon** [comment]
Ah, very true. In a future v2.0 we can default `noJunk` to `true`.

This PR needs a test. Please add some junk files to test/fixtures/before/custom_options/folder5/ and add `noJunk:true` to Gruntfile.js::`cleanempty.custom_options.options` ?

**Comment by ForestMist** [comment]
Done, tests pass on my local dev machine.

PS - Should folder5 be called folder4 to be in sequence with the existing folders 1, 2 and 3?

**Comment by stevenvachon** [comment]
folder4 should've already existed prior to running the test. I never wrote a reset function to regenerate the fixtures.

File: README.md [hunk 1]
```diff
@@ -55,6 +55,12 @@ Default value: `false`
 
 Override the task from blocking deletion of folders outside current working dir (CWD). Use with caution.
 
+#### options.noJunk
+Type: `Boolean`  
+Default value: `false`  
+
+Consider folders that only have [junk](https://github.com/sindresorhus/junk) files in them to be empty. Use in conjunction with options.folders set to true.
```
Comment (stevenvachon) [inline README.md]: `options.folders` and `true` need code back ticks

File: tasks/cleanempty.js [hunk 2]
```diff
@@ -39,7 +42,23 @@ module.exports = function(grunt)
...
+				var arr = fs.readdirSync(filepath);
+				var len = arr.length;
+				
+				if (options.noJunk && len > 0)
+				{
+					var allJunk = arr.every(function(name) 
+					{
+						return grunt.file.isFile(filepath + '/' + name) && junk.is(name);
+					});
+					if (!allJunk) continue;
```
Comment (stevenvachon) [inline tasks/cleanempty.js]: extra line break before this line. I'm anal

**Comment by stevenvachon** [comment]
Oh, and add some junk files to `test/fixtures/before/default_options/folder4/` and `test/fixtures/after/default_options/folder4/` as well. This will test that junk files are not removed when `noJunk` is falsey

**Comment by ForestMist** [comment]
Thanks for the all the details, changes uploaded and it is your project so be as anal as you like! :D

**Comment by stevenvachon** [comment]
:)

And you know what, folder4 was never committed/pushed to git because it was empty. I'll have to write a generator script in a future version to correct that.

Everything here looks good!

**Comment by stevenvachon** [comment]
v1.0.3 published. Thanks!

**Comment by ForestMist** [comment]
Most excellent. Thanks for all the PR guidance!

**Pull request merged**